documentation/DesignDocumentation.md

```diff
@@ -46,3 +69,155 @@
     - Team is unfamiliar with C# 
     - Proprietary
 - Due to some of the team's unfamiliarity with C# and the poor tooling available on macs, we will not opt for a Xamarin solution
+
+### 1.4 Library Usage
+The use of libraries is recommended for this project, and approved by the Product Owner. 
+
+**Justification**
+
+The main benefit of using existing code libraries is to save time on the project. A big priority is to get the application approved by the International Chess Federation (FIDA) as soon as possible, so saving time is a priority. 
+
+Furthermore, existing solutions in many cases might be more optimised than we could make from scratch with current resources. For example, there is no need to reinvent the wheel and code a chess engine when hundreds of such libraries are freely available, and are likely more optimised 
+
+Therefore, the use and adaptation of libraries is strongly encouraged where possible. 
+
+**Libraries Used**
+
+Currently, the following libraries have been investigated and are being used as part of the system architecture: 
+
+- **chess.ts**: A chess library and rewrite of chess.js in Typescript that is used for chess move generation/validation, piece placement/movement, and check/checkmate/stalemate detection.
```

Comment:
cann add that it needs modification to allow for illegal moves, also that we will wrap it in the tornello package so that it can be changed later